File: php/src/Advertisement/Domain/Errors/DescriptionTooLongError.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\Domain\Errors;

use Demo\App\Common\Result;

final readonly class DescriptionTooLongError
{
    public static function build(string $description): Result
    {
        return Result::failure(sprintf('Description has more than 200 characters: Has %d characters', mb_strlen($description)));
    }
}

File: kotlin/src/main/kotlin/advertisement/domain/errors/DescriptionTooLongError.kt
package advertisement.domain.errors

import common.domain.DomainError

class DescriptionTooLongError private constructor(errorMessage: String): DomainError(errorMessage) {
    companion object {
        fun withLongitudeMessage(value: String): DescriptionTooLongError {
            return DescriptionTooLongError("Description has more than 200 characters: Has ${value.length}")
        }
    }
}
